### DISCUSSION TITLE
Considering using Feature Management Infrastracture

### DISCUSSION THREAD

**Discussion by hishamco** [OrchardCMS/OrchardCore#11498] [Ideas]
I'm thinking from a while why we didn't add `OrchardCore.FeaturesManagement` that can built on top of `Microsoft.FeatureManagement` or extend it for our purpose, this will have a lot of advantages:

- Add feature management to the end users, this is handy for SAAS scenarios
- Modules feature can rebuilt on top this infrastracture
- A lot of configurations can be toggle on/off using the same infrastracture

**Comment by ns8482e** [comment]
@hishamco IMO both shares synergy.  the orchard’s implementation is better for multi tenancy and I guess you can also use `Microsoft.FeatureManagement` for feature flagging purpose within code in module pipeline

**Reply by ns8482e** [reply]
For example, tenant aware feature on controllers are defined using orchard’s feature attribute that allows tenant admin to manage from admin features UI

whereas, as developer, within that controller (A/B) feature flagging can be used to enable `hidden feature` via only  from config using `Microsoft.FeatureManagement`

**Reply by hishamco** [reply]
Thanks for your input Niraj, the intend is not necessary to replace everything with `Microsoft.FeaturesManagement` sure OC has good APIs for multitenancy as you mentioned but there are some similarities. What I'm asking for is have the goodness for both and push it into a thin layer that could be used in Features module as well as any other module that may required a features management either from business or UI perspective. Waiting for @jtkech thought
